pkg/provisioner/crdprovisioner.go

```diff
@@ -126,12 +127,36 @@ func (c *CrdProvisioner) CreateVolume(
 	secrets map[string]string,
 	volumeContentSource *v1alpha1.ContentVolumeSource,
 	accessibilityReq *v1alpha1.TopologyRequirement) (*v1alpha1.AzVolumeStatusParams, error) {
+	maxShares := 1
+	var err error
+	if maxSharesStr, ok := parameters["maxShares"]; ok {
+		maxShares, err = strconv.Atoi(maxSharesStr)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("parse %s failed with error: %v", maxSharesStr, err))
+		}
+		if maxShares < 1 {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("parse %s returned with invalid value: %d", maxSharesStr, maxShares))
+		}
+	}
+
+	maxMountReplicaCount := maxShares - 1
+	if maxMountReplicaCountStr, ok := parameters["maxMountReplicaCount"]; ok {
+		tempVal, err := strconv.Atoi(maxMountReplicaCountStr)
+		if err != nil {
+			klog.Warningf("setting default value derived from MaxShares: unable to parse MaxMountReplicaCount (%s): %v", maxMountReplicaCountStr, err)
+		}
+		if tempVal < maxMountReplicaCount {
+			maxMountReplicaCount = tempVal
+		}
+	}
```

Comment:
Should we return an `InvalidArgument` error if `maxMountReplicaCount >= `maxShares` instead of capping it `maxShares`?

Reply:
I wasn't sure of this. What would the right behavior be in such case? I feel like the volume creation should have some level of mistake coping and we can explain such behavior in our documentation.

Reply:
I fine either way as long as we document this behavior.